go.mod

```diff
@@ -15,13 +15,16 @@ require (
 	github.com/prometheus/exporter-toolkit v0.7.1
 	github.com/robfig/cron/v3 v3.0.1
 	github.com/spf13/pflag v1.0.5
+	github.com/stretchr/testify v1.7.0
 	golang.org/x/perf v0.0.0-20220317191433-96728ec1d780
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 	k8s.io/api v0.23.5
 	k8s.io/apimachinery v0.23.5
 	k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0
 	k8s.io/client-go v0.23.5
 	k8s.io/klog/v2 v2.60.0
 	k8s.io/sample-controller v0.23.4
+	sigs.k8s.io/controller-runtime v0.11.1
```

Comment:
Can we use client-go instead of controller-runtime? I think having both libraries in the same project can become challenging to maintain over time. It can also be confusing for contributors about which one to use.